students/douglas_klos/session3/mailroom.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+# Douglas Klos
+# January 28th, 2019
+# Python 210
+# mailroom.py
+
+# There are more functions than requested in the assignment.
+# I felt it needed a better way to add new donors than just through
+# send_thank_you function, and a way to add new donations to current
+# donors.  Note that you can add new donors through send_thank_you
+# as requested in the assignment.
+
+
+mailroom_db = [('Douglas', [5000, 2000]),
+               ('Maggie', [2222, 3333, 4444]),
+               ('Light Yagami', [124, 8975]),
+               ('Makise Kurisu', [235987]),
+               ('Youjo Senki', [13498, 9876, 1234]),
+               ('Motoko Kusanagi', [57892, 239857, 87265]),
+               ('Jo', [8814, 2320])
+               ]
+
+MAIN_PROMPT = '\n'.join((
+              '\nWelcome to the Mail Room!',
+              'Please choose from the following options: ',
+              '1: Send a Thank You',
+              '2: Create a report',
+              '3: Add a new donor',
+              '4: Add a new donation',
+              'q: Quit',
+              '>>> '))
+
+THANK_YOU_PROMPT = '\n'.join((
+                   '\nPlease enter one of the following:',
+                   '<Name>: Name of person to send a thank you note to',
+                   'list: Display donors in the database',
+                   'q: Return to main menu ',
+                   '>>> '))
+
+
+THANK_YOU_NOTE = '\n'.join((
```

Comment:
above was an example i gave to another student

Reply:
Yeah the join syntax was in the material.

So my gripe with using """ """ for these string literals is that I have to remove the visual indentation for the multiline string, as it would be included in the string itself.  Aesthetically I like the indent.

How about something like this, it gets away from the join syntax but keeps the visual indentation, and still passes flake8.

```
prompt = ('Welcome to mailroom 0.1!\n'
          'Please choose from the below options:\n'
          'mail - if you would like the mail menu.\n'
          'report - if you would like a report of donation totals.\n'
          'quit - Exit.\n'
          '>>> '
```
Thanks for the feedback

Reply:
Yep looks good.  The `join` is a clever solution, but I just don't find it very readable.  Articulating what you did there makes it very clear what is happening.